src/openrct2/world/map_animation.c

```diff
@@ -312,6 +312,8 @@ static bool map_animation_invalidate_track_onridephoto(sint32 x, sint32 y, sint3
 
     mapElement = map_get_first_element_at(x >> 5, y >> 5);
     do {
+        if (mapElement == NULL)
+            return false;
```

Comment:
I think it is supposed to return true if the animation should be removed from the animation list.